addon/utils/rdfa/event-processor.ts

```diff
@@ -128,24 +114,35 @@ export default EmberObject.extend({
    * @public
    */
   handleRegistryChange(/*registry*/) {
-    const editor = this.get('editor');
-    editor.clearHighlightForLocations(this.get('cardsLocationFlaggedRemoved'));
-
-    this.get('cardsLocationFlaggedNew').forEach(location => {
-      editor.highlightRange(location[0], location[1]);
-    });
+    let region;
+    while (region = this.cardsLocationFlaggedRemoved.shift()) {
```

Comment:
this was important as calling remove-highlight might in some cases trigger textual differences, if that happens `handleRegistryChange` will be called before this loop finishes. so need to make sure the card is removed from the array to prevent infinite loops.